lib/AppInfo/Application.php

```diff
@@ -21,51 +21,58 @@
-use OCA\GroupFolders\BackgroundJob\ExpireGroupPlaceholder;
-use OCA\GroupFolders\BackgroundJob\ExpireGroupTrash as ExpireGroupTrashJob;
-use OCA\GroupFolders\BackgroundJob\ExpireGroupVersions as ExpireGroupVersionsJob;
 use OCA\GroupFolders\CacheListener;
-use OCA\GroupFolders\Command\ExpireGroup\ExpireGroupBase;
-use OCA\GroupFolders\Command\ExpireGroup\ExpireGroupVersionsTrash;
-use OCA\GroupFolders\Command\ExpireGroup\ExpireGroupVersions;
-use OCA\GroupFolders\Command\ExpireGroup\ExpireGroupTrash;
-use OCA\GroupFolders\Folder\FolderManager;
+use OCP\AppFramework\IAppContainer;
+use OCA\GroupFolders\ACL\RuleManager;
 use OCA\GroupFolders\Helper\LazyFolder;
-use OCA\GroupFolders\Listeners\LoadAdditionalScriptsListener;
-use OCA\GroupFolders\Mount\MountProvider;
 use OCA\GroupFolders\Trash\TrashBackend;
 use OCA\GroupFolders\Trash\TrashManager;
-use OCA\GroupFolders\Versions\GroupVersionsExpireManager;
-use OCA\GroupFolders\Versions\VersionsBackend;
-use OCP\AppFramework\App;
-use OCP\AppFramework\Bootstrap\IBootContext;
+use OCA\GroupFolders\Mount\MountProvider;
+use OCA\GroupFolders\Folder\FolderManager;
 use OCP\AppFramework\Bootstrap\IBootstrap;
-use OCP\AppFramework\Bootstrap\IRegistrationContext;
-use OCP\AppFramework\IAppContainer;
 use OCP\AppFramework\Utility\ITimeFactory;
+use OCA\GroupFolders\ACL\ACLManagerFactory;
+use OCP\AppFramework\Bootstrap\IBootContext;
+use OCA\GroupFolders\Versions\VersionsBackend;
 use OCP\Files\Config\IMountProviderCollection;
-use OCP\ICacheFactory;
-use OCP\IDBConnection;
-use OCP\IGroup;
-use OCP\IConfig;
-use OCP\IGroupManager;
-use OCP\IRequest;
-use OCP\ISession;
-use OCP\IUserSession;
-use Psr\Log\LoggerInterface;
+use OCA\Files\Event\LoadAdditionalScriptsEvent;
+use OCA\GroupFolders\Service\DelegationService;
+use OCP\AppFramework\Bootstrap\IRegistrationContext;
+use OCA\GroupFolders\AuthorizedAdminSettingMiddleware;
+use OCA\Files_Sharing\Event\BeforeTemplateRenderedEvent;
+use OCA\GroupFolders\ACL\UserMapping\UserMappingManager;
+use OCA\GroupFolders\ACL\UserMapping\IUserMappingManager;
+use OCA\GroupFolders\Command\ExpireGroup\ExpireGroupBase;
+use OCA\GroupFolders\Versions\GroupVersionsExpireManager;
+use OCA\GroupFolders\BackgroundJob\ExpireGroupPlaceholder;
+use OCA\GroupFolders\Command\ExpireGroup\ExpireGroupTrash;
+use OCA\GroupFolders\Command\ExpireGroup\ExpireGroupVersions;
+use OCA\GroupFolders\Listeners\LoadAdditionalScriptsListener;
+use OCA\GroupFolders\Command\ExpireGroup\ExpireGroupVersionsTrash;
+use OCA\GroupFolders\BackgroundJob\ExpireGroupTrash as ExpireGroupTrashJob;
+use OCA\GroupFolders\BackgroundJob\ExpireGroupVersions as ExpireGroupVersionsJob;
 
 class Application extends App implements IBootstrap {
 	public function __construct(array $urlParams = []) {
 		parent::__construct('groupfolders', $urlParams);
 	}
 
+	public const CLASS_NAME_ADMIN_DELEGATION = 'OCA\GroupFolders\Settings\Admin';
+	public const APPS_USE_GROUPFOLDERS = [
+		'workspace'
+	];
```

Comment:
I thought it would be very interesting to have a const that list all the apps that would use Groupfolders. So, if you want to manipulate the interface or whatever on groupfolders. We cloud use ApplicationService and just add the name of our application to this const.

What do you think ?